packages/rs-drive-abci/src/abci/handlers.rs

```diff
@@ -686,22 +682,33 @@ where
         } else {
             let error = result.errors.first();
 
-            let error_message = if let Some(error) = error {
-                error.to_string()
+            let (code, error_message) = if let Some(error) = error {
+                match error {
+                    QueryError::NotFound(message) => {
+                        (AbciErrorCodes::NotFound as u32, message.to_owned())
+                    }
+                    QueryError::InvalidArgument(message) => {
+                        (AbciErrorCodes::InvalidArgument as u32, message.to_owned())
+                    }
+                    QueryError::Query(error) => {
+                        (AbciErrorCodes::InvalidArgument as u32, error.to_string())
+                    }
+                    _ => (AbciErrorCodes::Unknown as u32, error.to_string()),
```

Comment:
This must return en error here because we put something into validation result that we can't handle (programming error)

Reply:
WIll be fixed in the next PR